### PR TITLE
tests refactoring, new dataset added

### LINKED ISSUE #22: Add DICOM-RT test data
propose to include this:

http://neptun.phys.au.dk/~bassler/DKFZ/DICOM/

### PR DESCRIPTION
**Pull request by grzanka**
Major cleaning and revising of CI infrastructure and test suite.
Changes:
- added testing repository https://github.com/pytrip/testdata as git submodule to fix #22 
- moved tests (pytest call) from appveyor to github actions speedup testing procedure. 
- appveyor is responsible only for making and uploading binary wheel packages
- most important tests are marked as "@pytest.mark.smoke"
- very slow tests (>1 minute running time) are marked as "@pytest.mark.slow"
- on regular commit to a feature branch (with active PR) only short smoke tests will be triggered on github actions. Full tests will be executed on a commit to master branch (i.e. after merging PR) or on specific commit message [build]
- disabled codeql github actions workflow as nobody uses it
- updated .gitignore file
- fixed contributing guide
- moved pytriplib import (C extension) into method which are calling it. In this way most of the functionality of pytrip can be used without the need to compile the extension

### CODE REVIEW

**Review by rozchlastywacz — CHANGES_REQUESTED** [review]
Should testdata folder be empty?

File: .gitignore [hunk 1]
```diff
@@ -37,34 +40,100 @@ pip-delete-this-directory.txt
...
 
+# Django stuff:
+*.log
+local_settings.py
+db.sqlite3
+db.sqlite3-journal
+
+# Flask stuff:
+instance/
+.webassets-cache
```
[the remark this answers is not in the document]
Comment (grzanka) [reply .gitignore:67]: nope, I've used a "best practice" .gitignore from 
https://github.com/github/gitignore/blob/master/Python.gitignore

File: appveyor.yml [hunk 2]
```diff
@@ -9,9 +9,6 @@ environment:
      secure: GgEHENWgbfFj3Y3ESLvQqfHXKyp2BPy65U8B21ACBOw57Yv/epOgh8bp5tsi4PUa9ZkAIRW8m4GZx5HHm4q36zkXZZaZUZYV3qMRNpoGe/EweEdg8dqjW5wAGQ/bDQgVgfbWzg9o2KwuSFd+j1wLcKEc4ZMcKZvgOQlQQXqVKNB3VpeZftjI14ss1QPFgY4TLpa+bfaH1xHVCT/BKAy7y8wKBfZBvpb4xWOyOKaRKDa1+NOxS8TGYRBThWK6R9/gvXvh9ySzFwUbjCbHdcpK/w==
   PYPIUSER: __token__
   matrix:
-#    - platform: x64
-#      PYTHON: "C:\\Python27-x64"
-#
```
[the remark this answers is not in the document]
Comment (grzanka) [reply appveyor.yml:14]: it was already commented out, some tests were failing if I remember correctly
see https://github.com/pytrip/pytrip/issues/484
anyway we need a new issue to cross-check support for 2.x
probably sooner or later we should drop it

File: pytrip/dos.py [hunk 3]
```diff
@@ -108,10 +106,9 @@ def calculate_dvh(self, voi):
         for i in range(self.dimz):
             z_pos += self.slice_distance
             slice = voi.get_slice_at_pos(z_pos)
-            if slice is not None:   # VOI intersects with this slice
+            if slice is not None:  # VOI intersects with this slice
                 voi_and_cube_intersect = True
-                dose_bins += pytriplib.calculate_dvh_slice(self.cube[i],
-                                                           np.array(slice.contours[0].contour),
+                dose_bins += pytriplib.calculate_dvh_slice(self.cube[i], np.array(slice.contours[0].contour),
```
[the remark this answers is not in the document]
Comment (grzanka) [reply pytrip/dos.py:111]: I will check, btw - this was yapf not me :)

**Comment by grzanka** [comment]
> Should testdata folder be empty?

It looks empty, but it is a git submodule, to get its contents you need to clone the main repo with a special flag.
See https://stackoverflow.com/questions/3796927/how-to-git-clone-including-submodules

File: appveyor.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (grzanka) [reply appveyor.yml:14]: extracted to new issue https://github.com/pytrip/pytrip/issues/620

**Review by rozchlastywacz — APPROVED** [review]
Great job!

**Pull request merged**